api/controllers/EditTreatmentController.js

```diff
@@ -42,35 +76,88 @@ module.exports = {
       id: req.params.id,
       treatment: treatment,
       conditionList: conditionList,
-      oneValue: oneAttribute(conditionList),
-      medicalReport: req.session.medicalReport,
+      medicalReport: medicalReport,
     });
   },
 
-  update: function (req, res) {
-    const body = Object.assign({}, req.body);
-    delete body._csrf;
-
-    // use the value of the submit button to determine redirect
-    const action = body.save_and;
-
+  update: async function (req, res) {
     /**
-     * If there are newConditions, create them pre-validation
-     * and auto-select them
+     * Load the medical report, include the selected medication
      */
-    if (body.newConditions) {
-      req.body = ConditionHelper.addConditions(req, body, 'treatmentTreatedCondition');
+    let medicalReport = await MedicalReport.findOne({
+      where: {
+        applicationCode: req.session.applicationCode
+      },
+      include: [
+        {
+          model: Treatment,
+          as: 'Treatments',
+          where: { id: req.params.id }
+        }
+      ]
+    });
+
+    if (!medicalReport) {
+      // TODO: should probably flash a message 'condition not found'
+      return res.redirect(sails.route('treatments'));
     }
 
-    // the medications array is 0 indexed
-    const treatmentId = req.params.id - 1;
+    // use the value of the submit button to determine redirect
+    const action = req.body.save_and;
+
+    // the medication comes back as the first index of an array
+    let treatment = _.first(medicalReport.Treatments);
 
     let valid = req.validate(req, res, require('../schemas/treatment.schema'));
 
     if (valid) {
-      // replace the contents of the medication on the array
-      req.session.medicalReport.treatments[treatmentId] = body;
-      dataStore.storeMedicalReport(req.session.medicalReport);
+      treatment.update({
```

Comment:
Requested change: Similar to earlier comment on validating the input for existing conditions (i.e. the ones not in `newConditions`)